packages/ui-components/src/components/QrCode/QrCodeVisual.tsx

```diff
@@ -0,0 +1,203 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
+ */
+
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import {
+  Button,
+  Checkbox,
+  DialogActions,
+  FormControl,
+  FormControlLabel,
+  FormGroup,
+  Typography,
+  Grid,
+  Container
+} from '@material-ui/core';
+import CheckboxIcon from '@material-ui/icons/CheckBox';
+import DownloadIcon from '@material-ui/icons/GetApp';
+import { NoData } from '../NoData';
+import { QrCodeImage } from './QrCodeImage';
+import { getCanvasUrlForDownload } from './useQrCodeCanvas'
+
+const FormContainer = styled.div`
+  display: flex;
+  justify-content: center;
+  padding: 0px 40px 30px 40px;
+`;
+
+const Error = styled.div`
+  color: ${props => props.theme.palette.error.main};
+  margin-top: 10px;
+  text-align: center;
+`;
+
+const StyledCanvas = styled.canvas`
+  outline: 1px solid #dedede;
+  width: 500px;
+  height: 500px;
+`;
+
+interface DownloadQrCodeVisualProps {
+  downloadImages: (qrCodeCanvasUrlsWithFileNames: {name: string, url: string | undefined}[]) => Promise<void>;
+  config?: object;
```

Comment:
I'm not sure that `object` is the best `type` here. Technically it works, but I think we can be more descriptive. If we already have a config type for qr code viz in the `types` package you can use this, and if we don't I think it would be worth making one